packages/core/src/extension.ts

```diff
@@ -178,21 +179,27 @@ export async function activate(context: vscode.ExtensionContext) {
                             'OK'
                         )
                     } else {
-                        void vscode.window
-                            .showInformationMessage(
-                                'Amazon Q has moved to its own VSCode extension.' +
-                                    '\nInstall to work with Amazon Q, a generative AI assistant, with chat and code suggestions.',
-                                'Install',
-                                'Learn More'
-                            )
-                            .then(async resp => {
-                                if (resp === 'Learn More') {
-                                    // Clicking learn more will open the q extension page
-                                    await qExtensionPageCommand.execute()
-                                } else if (resp === 'Install') {
-                                    await installAmazonQExtension.execute()
-                                }
-                            })
+                        const dismissedInstall = globals.context.globalState.get<boolean>(amazonQInstallDismissedKey)
+                        if (!dismissedInstall) {
+                            void vscode.window
+                                .showInformationMessage(
+                                    'Amazon Q has moved to its own VSCode extension.' +
+                                        '\nInstall to work with Amazon Q, a generative AI assistant, with chat and code suggestions.',
+                                    'Install',
+                                    'Learn More'
+                                )
+                                .then(async resp => {
+                                    if (resp === 'Learn More') {
+                                        // Clicking learn more will open the q extension page
+                                        await qExtensionPageCommand.execute()
+                                    } else if (resp === 'Install') {
+                                        await installAmazonQExtension.execute()
+                                    } else if (resp === undefined) {
+                                        // when user click x button to dismiss it
+                                        await globals.context.globalState.update(amazonQInstallDismissedKey, true)
+                                    }
```

Comment:
maybe we shouldn't display the message again if users uninstall Q
```suggestion
                                    } else {
                                        await installAmazonQExtension.execute()
                                        await globals.context.globalState.update(amazonQInstallDismissedKey, true)
                                    }
```